src/Sarif/Visitors/VersionOneLogicalLocationKeyToFullyQualifiedNameMappingVisitor.cs

```diff
@@ -0,0 +1,77 @@
+    /// stores a mapping from fully qualified name to decorated name, if one exists. These data (the
+    /// decorated name and fully qualified name) moved from the location object to the logical location
+    /// object in v2.
+    /// </summary>
+    public class VersionOneLogicalLocationKeyToFullyQualifiedNameMappingVisitor : SarifRewritingVisitorVersionOne
+    {
+        public VersionOneLogicalLocationKeyToFullyQualifiedNameMappingVisitor()
+        {
+            FullyQualifiedNameToDecoratedNameMap = new Dictionary<string, string>();
+            LogicalLocationKeyToFullyQualifiedNameMap = new Dictionary<string, string>();
+        }
+        public IDictionary<string, string> FullyQualifiedNameToDecoratedNameMap { get; set; }
+
+        public IDictionary<string, string> LogicalLocationKeyToFullyQualifiedNameMap { get; set; }
+
+        public override LocationVersionOne VisitLocationVersionOne(LocationVersionOne node)
+        {
+            if (!string.IsNullOrEmpty(node.LogicalLocationKey) &&
+                !string.IsNullOrEmpty(node.FullyQualifiedLogicalName) &&
+                !node.FullyQualifiedLogicalName.Equals(node.LogicalLocationKey))
+            {
+                LogicalLocationKeyToFullyQualifiedNameMap[node.LogicalLocationKey] = node.FullyQualifiedLogicalName;
+            }
+
+            if (!string.IsNullOrEmpty(node.DecoratedName))
+            {
+                string fullyQualifiedName = node.FullyQualifiedLogicalName ?? node.LogicalLocationKey;
+                FullyQualifiedNameToDecoratedNameMap[fullyQualifiedName] = node.DecoratedName;
+            }
+
+            return base.VisitLocationVersionOne(node);
+        }
+
+        public override StackFrameVersionOne VisitStackFrameVersionOne(StackFrameVersionOne node)
+        {
+            if (!string.IsNullOrEmpty(node.LogicalLocationKey) &&
+                !string.IsNullOrEmpty(node.FullyQualifiedLogicalName) &&
+                !node.FullyQualifiedLogicalName.Equals(node.LogicalLocationKey))
+            {
+                LogicalLocationKeyToFullyQualifiedNameMap[node.LogicalLocationKey] = node.FullyQualifiedLogicalName;
+            }
+
+            // v1 stack frame does not reference a decorated name
+
+            return base.VisitStackFrameVersionOne(node);
+        }
+
+        public override AnnotatedCodeLocationVersionOne VisitAnnotatedCodeLocationVersionOne(AnnotatedCodeLocationVersionOne node)
+        {
+            if (!string.IsNullOrEmpty(node.LogicalLocationKey) &&
+                !string.IsNullOrEmpty(node.FullyQualifiedLogicalName) &&
+                !node.FullyQualifiedLogicalName.Equals(node.LogicalLocationKey))
+            {
+                LogicalLocationKeyToFullyQualifiedNameMap[node.LogicalLocationKey] = node.FullyQualifiedLogicalName;
+            }
+
+            // v1 stack frame does not reference a decorated name
```

Comment:
>stack frame [](start = 18, length = 11)

annotated code location.

Reply:
cut-n-paste ftw

---
In reply to: [244240816](https://github.com/Microsoft/sarif-sdk/pull/1185#discussion_r244240816) [](ancestors = 244240816)